### DISCUSSION TITLE
useGroupBy; initialState.groupBy; How to group by `accessor.nestedKey` instead of `accessor` itself?

### DISCUSSION THREAD

**Discussion by sthtnr** [TanStack/table#2620] [General] [answered]
I want to know how to group by a deeper hierarchy of key that `accessor` accesses, rather than by the value of `accessor` itself.
For example, suppose we have the following data:
```
let user = [
{
  "name": {
    "first": "aaa",
    "last": "bbb"
  },
  "age": 20
},
...
]
```
And if I set `name` as first column's accessor, Is it possible to group by `name.first` with `initialState.groupBy`? 
I tried like the following, but it didn't work.
```
initialState: {groupBy: ['name.first']}
```
```
initialState: {groupBy: [columns[0].accessor.first]}
```
I know I can accomplish to set `name.first` as accessor, but can't I accomplish this without doing so?

**Comment by gargroh** [comment, marked as answer]
Any operation like sort, filter or group will take place on `accessor` only, you can use `Cell` to format or show something else in display.

**Reply by sthtnr** [reply]
I see, understood. Thanks for your answer.
